File: LV 2/숫자의 표현.py
def solution(n):
    answer = 0
    start = 0
    finish = 1
    sumation = 1
    while True:
        if start == finish:
            break
        if n > sumation:
            finish += 1
            sumation += finish
        elif n == sumation:
            print(start, finish)
            answer += 1
            finish += 1
            sumation += finish
        else:
            start += 1
            sumation -= start
    
    return answer

File: LV 2/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int left = 0;
        int right = 0;
        int currentSum = 0;
        while (left <= right) {
            if (currentSum == n) {
                answer++;
            }

            if (currentSum <= n) {
                right++;
                currentSum += right;
            } else {
                currentSum -= left;
                left++;
            }
        }
        return answer;
    }
}
